oemof/solph/custom.py

```diff
@@ -839,3 +844,593 @@ def tes_ub_constr_rule(block, n, t):
             return (self.tes_level[n, t] <= n.params['tes_level_max'])
         self.tes_ub_constr = Constraint(
             self.GENERICCAES, m.TIMESTEPS, rule=tes_ub_constr_rule)
+
+
+########################################################################
+
+# ----------------------- DSM Component --------------------------------
+class SinkDSM(Sink):
+    r"""
+    Module for creating a Demand Side Management component.
+
+    Based on the paper by Zerrahn, Alexander and Schill, Wolf-Peter (2015):
+    On the representation of demand-side management in power system models,
+    in: Energy (84), pp. 840-845, 10.1016/j.energy.2015.03.037,
+    accessed 17.09.2019, pp. 842-843.
+
+    A special sink component which modifies the input demand series.
+
+    Parameters
+    ----------
+    demand: numeric
+        original electrical demand
+    capacity_up: int or array
+        maximum DSM capacity that may be increased
+    capacity_down: int or array
+        maximum DSM capacity that may be reduced
+    method: 'interval' , 'delay'
+
+        interval :
+
+            Simple model in which the load shift must be compensated in a
+            predefined fixed interval(**shift_interval is mandatory).
+            Foundation of this optimisation should be a potential analysis
+            of the DSM capacity during each interval. With this,
+            the boundaries for the DSM variable are set.
+
+        delay :
+
+            Sophisticated model based on the formulation by
+            Zerrahn & Schill (DIW).T he load-shift of the component must be
+            compensated in a predefined delay-time (**delay_time is mandatory).
+            DSM capacity can either be a fixed value or an hourly time series.
+    **shift_interval: int
+
+        only mandatory in method = 'interval' :
+
```

Comment:
"Mandatory if method == 'interval'" and "Only valid if method == 'interval'. For me "only mandatory" sounds to me that you can use it for 'delay' but it is not mandatory.